pkg/pipelinerunmetrics/metrics_test.go

```diff
@@ -504,6 +520,48 @@ func TestRecordRunningPipelineRunsCount(t *testing.T) {
 	metricstest.CheckLastValueData(t, "running_pipelineruns", map[string]string{}, 1)
 }
 
+func TestRecordRunningPipelineRunsCountAtPipelineLevel(t *testing.T) {
+	unregisterMetrics()
+
+	newPipelineRun := func(status corev1.ConditionStatus, namespace string) *v1.PipelineRun {
+		return &v1.PipelineRun{
+			ObjectMeta: metav1.ObjectMeta{Name: names.SimpleNameGenerator.RestrictLengthWithRandomSuffix("pipelinerun-"), Namespace: namespace},
+			Status: v1.PipelineRunStatus{
+				Status: duckv1.Status{
+					Conditions: duckv1.Conditions{{
+						Type:   apis.ConditionSucceeded,
+						Status: status,
+					}},
+				},
+			},
+		}
+	}
+
+	ctx, _ := ttesting.SetupFakeContext(t)
+	informer := fakepipelineruninformer.Get(ctx)
+	// Add N randomly-named PipelineRuns with differently-succeeded statuses.
+	for _, pipelineRun := range []*v1.PipelineRun{
+		newPipelineRun(corev1.ConditionTrue, "testns"),
+		newPipelineRun(corev1.ConditionUnknown, "testns"),
+		newPipelineRun(corev1.ConditionFalse, "testns"),
+	} {
+		if err := informer.Informer().GetIndexer().Add(pipelineRun); err != nil {
+			t.Fatalf("Adding TaskRun to informer: %v", err)
+		}
+	}
+
+	ctx = getConfigContextRunningPRLevel("pipeline")
+	recorder, err := NewRecorder(ctx)
+	if err != nil {
+		t.Fatalf("NewRecorder: %v", err)
+	}
+
+	if err := recorder.RunningPipelineRuns(informer.Lister()); err != nil {
+		t.Errorf("RunningPipelineRuns: %v", err)
+	}
+	metricstest.CheckLastValueData(t, "running_pipelineruns", map[string]string{"namespace": "testns", "pipeline": "anonymous"}, 1)
```

Comment:
The tags `namespace` and `pipeline` are set correctly, but the value `1` is only correct because the is only one running pipeline in the cluster. Adding more running pipelines, whether in the same namespace or others, causes the test to fail. This is because of the issue with the configuration highlighted in `metrics.go`.

I think it would be nice to have this as a matrix test which covers various possible scenarios, with configuration set and cluster, namespace, pipeline and pipelinerun level, and different initial inputs for each.

Reply:
will it be ok if I provide those tests in new PR ?